src/session/content/message_row/mod.rs

```diff
@@ -1,224 +1,115 @@
+mod base;
 mod media;
 mod media_picture;
 mod photo;
 mod sticker;
-mod sticker_paintable;
 mod text;
 
+use self::base::{MessageBase, MessageBaseExt, MessageBaseImpl};
 use self::media::Media;
 use self::media_picture::MediaPicture;
-pub(crate) use self::photo::MessagePhoto;
-pub(crate) use self::sticker::MessageSticker;
-use self::sticker_paintable::StickerPaintable;
-pub(crate) use self::text::MessageText;
+use self::photo::MessagePhoto;
+use self::sticker::MessageSticker;
+use self::text::MessageText;
 
-use gtk::prelude::*;
-use gtk::subclass::prelude::*;
-use gtk::{gdk, glib};
-
-use crate::session::chat::{Message, MessageForwardOrigin, MessageSender, SponsoredMessage};
-use crate::session::components::Avatar;
-use crate::session::ChatType;
-
-const AVATAR_SIZE: i32 = 32;
-const SPACING: i32 = 6;
-
-mod imp {
+#[gobject::gtk4_widget(final)]
+mod message_row {
```

Comment:
Clippy doesn't like that this mod has the same name as the containing module. We need to decide on which names we want to use for this kind of modules, since these will be present in each gobject implementation soon enough.

Reply:
Well, since the mod here represents our gobject implementation I think that it makes sense to use the gobject name, so I think I'll just deny this clippy lint for the entire project.